persistent-template/Database/Persist/TH.hs

```diff
@@ -1251,50 +1252,56 @@ maybeTyp :: Bool -> Type -> Type
 maybeTyp may typ | may = ConT ''Maybe `AppT` typ
                  | otherwise = typ
 
--- | produce code similar to the following:
+
+
+entityToPersistValueHelper :: (PersistEntity record) => record -> PersistValue
+entityToPersistValueHelper entity = PersistMap $ zip columnNames fieldsAsPersistValues
+    where
+        columnNames = map (unHaskellName . fieldHaskell) (entityFields (entityDef (Just entity)))
+        fieldsAsPersistValues = map toPersistValue $ toPersistFields entity
+
+entityFromPersistValueHelper :: (PersistEntity record) 
+                             => [String] -- ^ Column names, as '[String]' to avoid extra calls to "pack" in the generated code
```

Comment:
I don't know if this is a real optimization tbh. The previous code was using `String` already, maybe for this reason or maybe for no reason? It avoids `"pack" * numberOfFieldNames` is all